### PR TITLE
[Elastic Agent] Fix some typos

### PR DESCRIPTION
**Pull request by AndersonQ**
## What does this PR do?

It fixes some typos and Go conventions on the code's comments.

## Why is it important?

To keep consistent with the standard English.

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

**Review by ph — APPROVED** [review]

File: x-pack/elastic-agent/pkg/agent/application/gateway/fleet/fleet_gateway.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func newFleetGatewayWithScheduler(
 	stateStore stateStore,
 ) (gateway.FleetGateway, error) {
 
-	// Backoff implementation doesn't support the using context as the shutdown mechanism.
+	// Backoff implementation doesn't support the using of a context as the shutdown mechanism.
```
Comment (michel-laterman) [inline x-pack/elastic-agent/pkg/agent/application/gateway/fleet/fleet_gateway.go:130]:
```suggestion
	// Backoff implementation doesn't support the using cancellation of a context as the shutdown mechanism.
```
Is this what is meant?

**Review by michel-laterman — APPROVED** [review]
just a minor fix

File: x-pack/elastic-agent/pkg/agent/application/gateway/fleet/fleet_gateway.go (hunk 1 above)
Comment (efd6) [reply x-pack/elastic-agent/pkg/agent/application/gateway/fleet/fleet_gateway.go:130]: Suggest either "Backoff implementation doesn't support using a context [cancellation] as the shutdown mechanism." or "Backoff implementation doesn't support the use of a context [cancellation] as the shutdown mechanism." (insert "cancellation" according to taste).

**Review by ph — APPROVED** [review]

File: dev-tools/mage/crossbuild.go [hunk 2]
```diff
@@ -43,8 +43,8 @@ const defaultCrossBuildTarget = "golangCrossBuild"
 // See NewPlatformList for details about platform filtering expressions.
 var Platforms = BuildPlatforms.Defaults()
 
-// Types is the list of package types
-var SelectedPackageTypes []PackageType
+// SelectedPackageTypes is the list of package types
+var SelectedPackageTypes []PackageType = []PackageType{TarGz}
```
Comment (v1v) [inline dev-tools/mage/crossbuild.go:47]: Is this correct? The description mentioned comments only, but this is code?

I see failures in the packaging

Comment (v1v) [reply dev-tools/mage/crossbuild.go:47]: image

Comment (AndersonQ) [reply dev-tools/mage/crossbuild.go:47]: This was a mistake, but it's been fixed already by #30007.

**Pull request merged**